delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/RealmsDummy.scala

```diff
@@ -0,0 +1,134 @@
+
+  override def fetchAt(label: Label, rev: Long): IO[RealmRejection.RevisionNotFound, Option[RealmResource]] =
+    stateAt(label, rev).map(_.flatMap(_.toResource))
+
+  override def list(pagination: FromPagination, params: RealmSearchParams): UIO[UnscoredSearchResults[RealmResource]] =
+    cache.get.hideErrors.map { resources =>
+      val filtered = resources.values
+        .filter { resource =>
+          params.rev.forall(_ == resource.rev) &&
+          params.deprecated.forall(_ && resource.deprecated) &&
+          params.createdBy.forall(_ == resource.createdBy) &&
+          params.updatedBy.forall(_ == resource.updatedBy)
+        }
+        .toVector
+        .sortBy(_.createdAt)
+      UnscoredSearchResults(
+        filtered.length.toLong,
+        filtered.map(UnscoredResultEntry(_)).slice(pagination.from, pagination.from + pagination.size)
+      )
+    }
+
+  private def setToCache(resource: RealmResource): UIO[RealmResource]         =
+    cache.update(_ + (resource.id -> resource)).hideErrors.as(resource)
+
+  private def deprecateFromCache(resource: RealmResource): UIO[RealmResource] =
+    cache.update(_.updatedWith(resource.id)(_.as(resource))).hideErrors.as(resource)
+
+  private def currentState(label: Label): UIO[Option[RealmState]] =
+    journal.get.hideErrors.map { labelsEvents =>
+      labelsEvents.get(label).map(_.foldLeft[RealmState](Initial)(Realms.next))
+    }
+
+  private def stateAt(label: Label, rev: Long): IO[RevisionNotFound, Option[RealmState]] =
+    journal.get.hideErrors.flatMap { labelsEvents =>
+      labelsEvents.get(label).traverse { events =>
+        if (events.size < rev)
+          IO.raiseError(RevisionNotFound(rev, events.size.toLong))
+        else
+          events
+            .foldLeft[RealmState](Initial) {
+              case (state, event) if event.rev <= rev => Realms.next(state, event)
+              case (state, _)                         => state
+            }
+            .pure[UIO]
+      }
+    }
+
+  private def eval(cmd: RealmCommand): IO[RealmRejection, RealmResource] =
+    semaphore.withPermit {
+      for {
+        lbEvents <- journal.get
+        state     = lbEvents.get(cmd.label).fold[RealmState](Initial)(_.foldLeft[RealmState](Initial)(Realms.next))
+        event    <- Realms.evaluate(wellKnown)(state, cmd)
+        _        <- journal.set(lbEvents.updatedWith(cmd.label)(_.fold(Some(Vector(event)))(events => Some(events :+ event))))
+        res      <- IO.fromEither(Realms.next(state, event).toResource.toRight(UnexpectedInitialState(cmd.label)))
+      } yield res
+    }
+}
+
+object RealmsDummy {
+
+  type RealmsJournal = Map[Label, Vector[RealmEvent]]
+  type RealmsCache   = Map[Label, RealmResource]
+
+  /**
+    * Creates a new dummy Realms implementation.
+    *
+    * @param wellKnown the well known configuration for an OIDC provider resolver
+    */
+  final def apply(wellKnown: WellKnownResolver)(implicit clock: Clock[UIO] = IO.clock): UIO[RealmsDummy] =
```

Comment:
I think that maybe it would be worth adding an overload that takes the wellKnown Uri to WellKnown mapping and construct the realms dummy.